### ISSUE TITLE
Sudo password on app/mon can't have spaces

### ISSUE THREAD

**Issue by redshiftzero** [freedomofpress/securedrop#2013]
## Description

When installing SecureDrop, if you set up the user on app and mon to require a sudo password with spaces, an authentication error occurs due to early termination of the password string

## Steps to Reproduce

1. `./securedrop-admin sdconfig`
2. You have a sudo password on app and mon with spaces, e.g. `word1 word2`
3. Run `./securedrop-admin install`

## Expected Behavior

Install completes without problems.

## Actual Behavior

Install fails with an error as follows:

```
TASK [setup] *******************************************************************
fatal: [mon]: FAILED! => {"failed": true, "msg": "Incorrect sudo password"}
fatal: [app]: FAILED! => {"failed": true, "msg": "Incorrect sudo password"}
	to retry, use: --limit @/home/amnesia/Persistent/securedrop/install_files/ansible-base/securedrop-prod.retry

[...snip!...]

Traceback (most recent call last):
  File "./securedrop-admin", line 262, in <module>
    args.func(args)
  File "./securedrop-admin", line 216, in install_securedrop
    cwd=ANSIBLE_PATH)
  File "/usr/lib/python2.7/subprocess.py", line 186, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/home/amnesia/Persistent/securedrop/./install_files/ansible-base/securedrop-prod.yml', '-e ansible_become_pass=word1 word2']' returned non-zero exit status 2
```

## Comments

Simple fix: add quotes around the password when constructing the arguments for `subprocess.check_call` in `install_securedrop`

**Comment by conorsch** [comment]
Great catch in QA, @redshiftzero. While testing #2014, I tried a few other creative password types, and was able to find problems with handling special characters of several kinds, not just whitespace. For example, using the fix in #2014, an exception will still be thrown if the password contains a single quote. That's bad. 

All this goes to show that we're going about the problem wrong, given that we need to worry about passing the passphrase around at all. Since we're already prompting for the sudo password every time `./securedrop-admin install` is run, we should just be able to pass `--ask-become-pass` to the playbook command and let Ansible handle the sudo password appropriately. Even if blank, it should work as it does now.

We should also add some guiding language to the logging output to be clearer about whether a sudo password is necessary—the prompt is currently quite confusing:

```
$ ./securedrop-admin install
INFO: Now installing SecureDrop on remote servers.
Sudo password for servers [blank if not needed]:
```

How does an Admin determine "if not needed"? We can improve it to something like:

```
$ ./securedrop-admin install
INFO: Now installing SecureDrop on remote servers.
INFO: You will be prompted for the sudo password on the servers.
INFO: The sudo password is only necessary during initial installation.
Sudo password for servers [blank if not needed]:
```

We could technically detect whether this is an initial install by checking for `app-ssh-aths` on disk, but that seems more convoluted and inappropriate for a QA fix.

**Comment by redshiftzero** [comment]
Ah yep, great points. That's a much better solution, I'll test `--ask-become-pass` on a hardware instance and update #2014. And I agree, this should be scoped to just preventing the confusing breakage. We can improve further in a future release.

**Comment by conorsch** [comment]
Resolved via #2014.
